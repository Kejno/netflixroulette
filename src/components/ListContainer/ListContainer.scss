.filterWrapp {
  display: flex;
  justify-content: space-between;
}
