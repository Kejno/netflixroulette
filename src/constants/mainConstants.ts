export const BASE_URL: string = 'http://localhost:4000';

export const tabSortOptions: any = {
  sort: ['release_date', 'vote_average'],
  tab: {
    all: null,
    documentary: 'documentary',
    comedy: 'comedy',
    horror: 'horror',
    crime: 'crime',
  },
  sortOrder: ['asc', 'desc'],
};

export const initialState = {
  movieList: {
    movies: [{}],
  },
  settings: {
    sortBy: 'release_date',
    filter: '',
    sortOrder: 'desc',
  },
  search: {
    searchBy: 'title',
    search: '',
  },
  movieDetails: {
    movie: {
      genres: [''],
    },
  },
  movieDelete: {},
  movieUpdate: {
    movie: {},
  },
  movieCreate: {},
  isVisibleSearch: {
    isVisible: false,
  },
};
