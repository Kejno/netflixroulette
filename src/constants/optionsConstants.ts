export const NEW_SORT_OPTIONS = 'NEW_SORT_OPTIONS';
export const NEW_FILTER_OPTIONS = 'NEW_FILTER_OPTIONS';
