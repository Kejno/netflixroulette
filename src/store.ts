import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import {
  searchReducer,
  isVisibleSearchBarReducer,
} from './reducers/searchReducer';
import {
  movieListReducer,
  movieDeleteReducer,
  movieDetailsReducer,
  movieUpdateReducer,
  movieCreateReducer,
} from './reducers/movieReducer';
import { settingsReducer } from './reducers/settingsReducer';

const middleware = [thunk];

export const reducer = combineReducers({
  movieList: movieListReducer,
  settings: settingsReducer,
  search: searchReducer,
  movieDetails: movieDetailsReducer,
  movieDelete: movieDeleteReducer,
  movieUpdate: movieUpdateReducer,
  movieCreate: movieCreateReducer,
  isVisibleSearch: isVisibleSearchBarReducer,
});

const InitialState = () => {
  return {};
};

export const initialState = {};

const store = createStore(
  reducer,
  initialState,
  applyMiddleware(...middleware)
);
store.subscribe(() => {
  console.log('Subscribe', store.getState());
});

export default store;
