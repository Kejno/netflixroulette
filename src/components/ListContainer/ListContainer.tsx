import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getMovies, movieDetails } from '../../actions/movieActions';
import { tabSortOptions } from '../../constants/mainConstants';
import {
  changeSortOtps,
  changeFilterOpts,
} from '../../actions/settingsActions';
import { isVisibleSearchBar } from '../../actions/searchActions';
import Loader from '../Loader/Loader';
import Sort from '../Sort/Sort';
import Tab from '../Tab/Tab';
import MovieList from '../MovieList/MovieList';

import cls from './ListContainer.scss';

const ListContainer = () => {
  const dispatch = useDispatch();

  const movieList = useSelector((state: any) => state.movieList);
  const optionsList = useSelector((state: any) => state.settings);
  const searchParams = useSelector((state: any) => state.search);

  const { loading, error, movies } = movieList;

  const { data = [] } = movies;
  const { tab } = tabSortOptions;

  useEffect(() => {
    dispatch(getMovies({ ...optionsList, ...searchParams }));
  }, [optionsList]);

  const getMovieById = (id: number): void => {
    dispatch(movieDetails(id));
    dispatch(isVisibleSearchBar());
  };

  return (
    <>
      <div className={cls.filterWrapp}>
        <Tab
          onChangeTab={(key) => {
            dispatch(changeFilterOpts(Object.values(tab)[key]));
          }}
          activeTab={optionsList.filter}
        />
        <Sort
          onChange={(e) => {
            dispatch(changeSortOtps(e.target.value));
          }}
        />
      </div>

      {loading ? (
        <Loader />
      ) : error ? (
        <div>Error</div>
      ) : (
        <MovieList movies={data} getMovieById={getMovieById} />
      )}
    </>
  );
};

export default ListContainer;
