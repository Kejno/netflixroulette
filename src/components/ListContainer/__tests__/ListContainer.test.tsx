import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import { initialState as fakeStore } from '../../../constants/mainConstants';
import { createStore } from 'redux';
import ListContainer from '../ListContainer';
import { reducer } from '../../../store';

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('ListContainer component', () => {
  let getWrapper: any;
  let mockStore: any;

  beforeEach(() => {
    mockStore = createStore(reducer, fakeStore);
    getWrapper = () =>
      mount(
        <Provider store={mockStore}>
          <ListContainer />
        </Provider>
      );
  });

  describe('Test Snapshot & Components', () => {
    it('should match snapshot', () => {
      mockStore.dispatch = jest.fn();
      const wrapper = getWrapper(mockStore);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should display tag div', () => {
    mockStore.dispatch = jest.fn();
    const wrapper = getWrapper(mockStore).find('div');
    expect(wrapper.length).toBe(2);
  });
});
