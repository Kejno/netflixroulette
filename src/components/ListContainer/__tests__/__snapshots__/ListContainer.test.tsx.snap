// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer component Test Snapshot & Components should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ListContainer>
    <div
      className="filterWrapp"
    >
      <Tab
        activeTab=""
        onChangeTab={[Function]}
      >
        <ul
          className="tabContainer"
        >
          <li
            className="tabItem false"
            key="0"
            onClick={[Function]}
          >
            ALL
          </li>
          <li
            className="tabItem false"
            key="1"
            onClick={[Function]}
          >
            DOCUMENTARY
          </li>
          <li
            className="tabItem false"
            key="2"
            onClick={[Function]}
          >
            COMEDY
          </li>
          <li
            className="tabItem false"
            key="3"
            onClick={[Function]}
          >
            HORROR
          </li>
          <li
            className="tabItem false"
            key="4"
            onClick={[Function]}
          >
            CRIME
          </li>
        </ul>
      </Tab>
      <Sort
        onChange={[Function]}
      >
        <select
          className="selectSort"
          name="calc"
          onChange={[Function]}
        >
          <option
            className="optionSort"
            key="0"
            value="release_date"
          >
            RELEASE DATE
          </option>
          <option
            className="optionSort"
            key="1"
            value="vote_average"
          >
            VOTE AVERAGE
          </option>
        </select>
      </Sort>
    </div>
    <MovieList
      getMovieById={[Function]}
      movies={Array []}
    >
      <div
        className="movieListContainer"
      />
    </MovieList>
  </ListContainer>
</Provider>
`;
